marine.c

```diff
@@ -12,161 +12,65 @@
 
 // TODO remove unused imports
 // TODO remove Windows ifdefs, we're not going to support Windows for now.
-// TODO find a good way to write tests for performance, accuracy and memory leaks in the C code.
+// TODO find a good way to write tests for performance, accuracy and memory
+// leaks in the C code.
 #include "marine.h"
-
-#include <config.h>
-
+#include "config.h"
+#include <glib.h>
+#include <limits.h>
+#include <locale.h>
 #include <stdlib.h>
-#include <stdio.h>
 #include <string.h>
-#include <locale.h>
-#include <limits.h>
-
-#ifdef HAVE_GETOPT_H
-
-#include <getopt.h>
-
-#endif
-
-#include <errno.h>
 
 #ifdef _WIN32
-# include <winsock2.h>
+#include <winsock2.h>
```

Comment:
If you're fixing up includes, might as-well remove this one (since it's under `ifdef _WIN32`) (and maybe there are others like this)

Reply:
I don't really mind about keeping it, it doesn't bother us right now and might save us some time if we'd ever like to support windows.
If you think it's confusing and doesn't worth it, reply and I'll remove it.